File: client/bindings/python/native/examples/ledger_nano.py
from iota_client import IotaClient, LedgerNanoSecretManager

# In this example we will get the ledger status and generate an address
# To use the ledger nano simulator clone https://github.com/iotaledger/ledger-shimmer-app, run `git submodule init && git submodule update --recursive`,
# then `./build.sh -m nanos|nanox|nanosplus -s` and use `True` in `LedgerNanoSecretManager(True)`.

# Create an IotaClient instance
client = IotaClient({'nodes': ['https://api.testnet.shimmer.network']})

is_simulator = True

secret_manager = LedgerNanoSecretManager(is_simulator)

# Get the Ledger Nano status.
ledger_nano_status = client.get_ledger_nano_status(is_simulator)

print(f'Ledger Nano status: {ledger_nano_status}')

# Generate public address with custom account index and range.
address = client.generate_addresses(secret_manager, {
    "accountIndex": 0,
    "range": {
        "start": 0,
        "end": 1,
    },
})

print(f'Address: {address[0]}')

File: client/bindings/nodejs/examples/ledger_nano.ts
// Copyright 2021-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Client, initLogger } from '@iota/client';
require('dotenv').config({ path: '../.env' });

// Run with command:
// node ./dist/ledger_nano.js

// In this example we will get the ledger status and generate an address
// To use the ledger nano simulator clone https://github.com/iotaledger/ledger-shimmer-app, run `git submodule init && git submodule update --recursive`,
// then `./build.sh -m nanos|nanox|nanosplus -s` and use `true` in `LedgerSecretManager::new(true)`.
async function run() {
    initLogger();
    if (!process.env.NODE_URL) {
        throw new Error('.env NODE_URL is undefined, see .env.example');
    }

    const client = new Client({
        // Insert your node URL in the .env.
        nodes: [process.env.NODE_URL],
    });

    try {
        const isSimulator = false;

        const secretManager = { ledgerNano: isSimulator };

        const ledgerNanoStatus = await client.getLedgerNanoStatus(isSimulator);

        console.log(ledgerNanoStatus);

        const address = await client.generateAddresses(secretManager, {
            accountIndex: 0,
            range: {
                start: 0,
                end: 1,
            },
        });

        console.log('First public address:', address, '\n');
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
